src/font.js

```diff
@@ -274,10 +267,14 @@ Font.prototype.getKerningValue = function(leftGlyph, rightGlyph) {
  */
 Font.prototype.defaultRenderOptions = {
     kerning: true,
-    features: {
-        liga: true,
-        rlig: true
-    }
+    features: [
+        /**
+         * these 4 features are required to render Arabic text properly
+         * and shouldn't be turned off when rendering arabic text.
+         */
+        { script: 'arab', tags: ['init', 'medi', 'fina', 'rlig'] },
+        { script: 'latn', tags: ['liga', 'rlig'] }
```

Comment:
So if the font is not detected as Latin `liga` & `rlig` will be off?
Like if a font only supports emoji or a certain script?

Reply:
**Yes**, every registered feature is checked before application. If the **script is missing** or the font **doesn't support this feature**, It'll silently fail to apply this feature.